pkg/genericapiserver/config_selfclient.go

```diff
@@ -107,6 +129,41 @@ func (s *SecureServingInfo) NewSelfClientConfig(token string) (*restclient.Confi
 	return clientConfig, nil
 }
 
+func trustedChain(chain []*x509.Certificate) bool {
+	intermediates := x509.NewCertPool()
+	for _, cert := range chain[1:] {
+		intermediates.AddCert(cert)
+	}
+	_, err := chain[0].Verify(x509.VerifyOptions{
+		Intermediates: intermediates,
+		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+	})
```

Comment:
This only checks the trust, not the names or IPs. The later is checked in the calling code in addition.